### PR TITLE
[SPARK-31950][SQL][TESTS] Extract SQL keywords from the generated parser class

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR intends to extract SQL keywords (`allCandidateKeywords` in `TableIdentifierParserSuite`) from the generated parser class.

### Why are the changes needed?

It is hard to maintain a full set of SQL keywords in `TableIdentifierParserSuite`, so it would be nice if we could update `allCandidateKeywords` automatically. 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala [hunk 1]
```diff
@@ -562,6 +376,7 @@ class TableIdentifierParserSuite extends SparkFunSuite with SQLHelper {
     "except",
     "false",
     "fetch",
+    "filter",
```
Comment (maropu) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala:379]: I noticed that we forgot to add this word in this list by this change.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala [hunk 2]
```diff
@@ -285,255 +287,67 @@ class TableIdentifierParserSuite extends SparkFunSuite with SQLHelper {
...
-    "values",
-    "view",
-    "views",
-    "when",
-    "where",
-    "window",
-    "with",
-    "year")
+  // All the SQL keywords defined in `SqlBase.g4`
+  val allCandidateKeywords = {
```
Comment (maropu) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala:291]: Note: I've checked that (`the existing allCandidateKeywords` -- `this new allCandidateKeywords`) is empty.

**Comment by maropu** [comment]
How about this update? @cloud-fan @viirya @dilipbiswal

**Comment by viirya** [comment]
Text-based solution sounds good.

**Comment by maropu** [comment]
https://github.com/apache/spark/pull/28802

**Pull request closed without merging**